File: Part1_강의자료/Ch07_이분탐색/문제별코드/1920_수찾기/1920.cpp
#include <algorithm>
#include <stdio.h>
#include <vector>
using namespace std;

int is_exist(int N, vector<int> &vec, int A)
{
    int left = 0, right = N - 1, mid;

    while (left <= right)
    {
        mid = (left + right) / 2;
        if (vec[mid] < A)
            left = mid + 1;
        else if (vec[mid] > A)
            right = mid - 1;
        else
            return 1;
    }
    return 0;
}

int main()
{
    int N, M;

    scanf("%d", &N);
    vector<int> vec(N);

    for (int i = 0; i < N; i++)
        scanf("%d", &vec[i]);

    sort(vec.begin(), vec.end());

    int A;
    scanf("%d", &M);
    for (int i = 0; i < M; i++)
    {
        scanf("%d", &A);
        // printf("%d\n", binary_search(vec.begin(), vec.end(), A));
        printf("%d\n", is_exist(N, vec, A));
    }

    printf("\n");
    return 0;
}

File: Part1_강의자료/Ch07_이분탐색/문제별코드/1920_수찾기/1920.py
N = int(input())
li = sorted(list(map(int,input().split())))

def is_exist(A):
    left, right = 0, N - 1
    while left <= right:
        mid = (left + right) // 2
        if li[mid] < A:
            left = mid + 1
        elif li[mid] > A:
            right = mid - 1
        else:
            return 1
    return 0

M = int(input())
for i in map(int,input().split()):
    print(is_exist(i))
